### PR TITLE
Rust: Refactor Tags To Lanes

### PR DESCRIPTION
**Pull request by droogmic**
Preparation for more business logic to be added.

I could start with busses next, but I think I would rather see what happens when I try to add some separators

### CODE REVIEW

**Comment by droogmic** [comment]
I did however, royally screw up when picking which commit to refactor, so need to redo the work...

File: rust/osm2lanes-web/src/main.rs [hunk 1]
```diff
@@ -8,10 +8,9 @@ use yew::prelude::*;
 
 mod draw;
 
-use osm2lanes::{
-    lanes_to_tags, tags_to_lanes_with_warnings, DrivingSide, Lane, LanePrintable, Lanes,
-    LanesToTagsConfig, Locale, Tags,
-};
+use osm2lanes::{lanes_to_tags, tags_to_lanes, LanesToTagsConfig, TagsToLanesConfig};
```
Comment (dabreegster) [inline rust/osm2lanes-web/src/main.rs:11]: Curious what motivates this split? If the namespace feels like it's getting too big, maybe we shouldn't publish a flattened one, but have modules maybe?

File: rust/osm2lanes/src/lib.rs [hunk 2]
```diff
@@ -119,7 +119,7 @@ impl LanePrintable for Lane {
                 designated: LaneDesignated::Bus,
                 ..
             } => '🚌',
-            Self::Shoulder => 'ˢ',
+            Self::Shoulder => '🛆',
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:122]: Hmm, my browser and terminal both have trouble with this. Might just be an issue with my system

File: rust/osm2lanes/src/transform/tags_to_lanes/bicycle.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+pub fn bicycle(
+    tags: &Tags,
+    locale: &Locale,
+    oneway: bool,
+    forward_side: &mut Vec<Lane>,
+    backward_side: &mut Vec<Lane>,
+    warnings: &mut RoadWarnings,
+) -> ModeResult {
+    impl Tags {
```
Comment (dabreegster) [inline rust/osm2lanes/src/transform/tags_to_lanes/bicycle.rs:11]: Rust is crazy -- I didn't realize you could add a method to a struct just in the middle of some other method! Any reason to not put this outside of `fn bicycle` and just in this module scope?

**Review by dabreegster — APPROVED** [review]
Giant files being split always good :)

File: rust/osm2lanes-web/src/main.rs (hunk 1 above)
Comment (droogmic) [reply rust/osm2lanes-web/src/main.rs:11]: I'll be honest, I don't know how to do this properly in rust and have not yet looked it up.

Maybe I'll make it the topic of my next refactoring :)

File: rust/osm2lanes/src/lib.rs (hunk 2 above)
Comment (dabreegster) [reply rust/osm2lanes/src/lib.rs:122]: I'm fine with this one, I'll see if I can install that font...

File: rust/osm2lanes/src/transform/tags_to_lanes/bicycle.rs (hunk 3 above)
Comment (droogmic) [reply rust/osm2lanes/src/transform/tags_to_lanes/bicycle.rs:11]: When I implemented this, I assumed this falls under the general programming "smallest scope is better".

In this case though after having copied this function into its own file, I agree this impl should sit next to the function

**Pull request merged**